content/en/blog/_posts/clientcmd-tutorial.md

```diff
@@ -0,0 +1,292 @@
+
+The "deferred" mention in these function names refers to the fact that
+the final configuration will be determined as late as possible.
+This means that these functions can be called before the command line arguments are parsed,
+and the resulting configuration will use whatever values have been parsed
+by the time it's actually constructed.
+
+### Obtain an API client
+
+The merged configuration is returned as a
+[`ClientConfig`](https://pkg.go.dev/k8s.io/client-go/tools/clientcmd#ClientConfig) instance.
+An API client can be obtained from that by calling the `ClientConfig()` method.
+
+If no configuration is given
+(`KUBECONFIG` is empty or points to non-existent files,
+`~/.kube/config` doesn't exist,
+and no configuration is given using command line arguments),
+the default setup will return an obscure error referring to `KUBERNETES_MASTER`.
+This is legacy behaviour;
+several attempts have been made to get rid of it,
+but it is preserved for the `--local` and `--dry-run` command line arguments in `--kubectl`.
+You should check for "empty configuration" errors by calling `clientcmd.IsEmptyConfig()`
+and provide a more explicit error message.
+
+The `Namespace()` method is also useful:
+it returns the namespace that should be used.
+It also indicates whether the namespace was overridden by the user
+(using `--namespace`).
+
+## Full example
+
+Here's a complete example.
+
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/spf13/pflag"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func main() {
+	// Loading rules, no configuration
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+
+	// Overrides and flag (command line argument) setup
+	configOverrides := &clientcmd.ConfigOverrides{}
+	flags := pflag.NewFlagSet("clientcmddemo", pflag.ExitOnError)
+	clientcmd.BindOverrideFlags(configOverrides, flags,
+		clientcmd.RecommendedConfigOverrideFlags(""))
+	flags.StringVarP(&loadingRules.ExplicitPath, "kubeconfig", "", "", "absolute path(s) to the kubeconfig file(s)")
+	flags.Parse(os.Args)
+
+	// Client construction
+	kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+	config, err := kubeConfig.ClientConfig()
+	if err != nil {
+		if clientcmd.IsEmptyConfig(err) {
+			panic("Please provide a configuration pointing to the Kubernetes API server")
+		}
+		panic(err)
+	}
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err)
+	}
+
+	// How to find out what namespace to use
+	namespace, overridden, err := kubeConfig.Namespace()
+	if err != nil {
+		panic(err)
+	}
+	fmt.Printf("Chosen namespace: %s; overridden: %t\n", namespace, overridden)
+
+	// Let's use the client
+	nodeList, err := client.CoreV1().Nodes().List(context.TODO(), v1.ListOptions{})
+	if err != nil {
+		panic(err)
+	}
+	for _, node := range nodeList.Items {
+		fmt.Println(node.Name)
+	}
+}
+```
```

Comment:
I'd wrap up the article with a next step, even if that's as simple as “Happy coding!”.